File: src/util/GameSettings.java
package util;

/**
 * A simple structure containing the size of the board and
 * the minimum sequence/streak length required to win.
 * You should not change this class.
 *
 * @author David Symons
 */
public final class GameSettings
{
	public final int nrRows;
	public final int nrCols;
	public final int minStreakLength; // The minimum sequence length required to win.
	
	// Loads the default settings if no arguments are provided.
	public GameSettings()
	{
		this(6, 7, 4);
	}
	
	// Creates an instance with the given settings.
	public GameSettings(int nrRows, int nrCols, int minStreakLength)
	{
		this.nrRows = nrRows;
		this.nrCols = nrCols;
		this.minStreakLength = minStreakLength;
	}
}

File: ConnectX TS/util/GameSettings.ts
export class GameSettings{
    public nrRows: number
    public nrCols: number
    public minStreakLength: number

    // public GameSettings(){
    //     this.nrRows = 6;
    //     this.nrCols = 7;
    //     this.minStreakLength = 4;
    // }

    public GameSettings(rows:number, cols:number, streakLength:number){
        this.nrRows = rows
        this.nrCols = cols
        this.minStreakLength = streakLength
    }
}